src/soundfile.js

```diff
@@ -62,102 +271,1550 @@ const ac = p5sound.audiocontext;
  *  }
  * </code></div>
  */
-p5.SoundFile = function (paths, onload, onerror, whileLoading) {
-  if (typeof paths !== 'undefined') {
-    if (typeof paths === 'string' || typeof paths[0] === 'string') {
-      var path = p5.prototype._checkFileFormats(paths);
-      this.url = path;
-    } else if (typeof paths === 'object') {
-      if (
-        !(window.File && window.FileReader && window.FileList && window.Blob)
-      ) {
-        // The File API isn't supported in this browser
-        throw 'Unable to load file because the File API is not supported';
+class SoundFile {
+  constructor(paths, onload, onerror, whileLoading) {
+    if (typeof paths !== 'undefined') {
+      if (typeof paths === 'string' || typeof paths[0] === 'string') {
+        var path = p5.prototype._checkFileFormats(paths);
```

Comment:
something for the backlog: we should probably import `checkFileFormats` and remove it from the p5 prototype

Reply:
looks like we've started taking care of this in https://github.com/processing/p5.js-sound/pull/540, fantastic!